src/expand.cpp

```diff
@@ -189,9 +189,14 @@ static size_t parse_slice(const wchar_t *in, wchar_t **end_ptr, std::vector<long
 
         const wchar_t *end;
         long tmp = fish_wcstol(&in[pos], &end);
-        // We don't test `*end` as is typically done because we expect it to not be the null char.
-        // Ignore the case of errno==-1 because it means the end char wasn't the null char.
-        if (errno > 0) {
+        if (idx.empty() && in[pos] == L'.' && in[pos + 1] == L'.') {
```

Comment:
This is confusing to read - we call `wcstol` but sometimes ignore the result. Can we rearrange the test so that we check for a missing index before calling `wcstol()`?

Reply:
Yep, done!